Source/ViewControllers/BrickCollectionView.swift

```diff
@@ -660,23 +661,34 @@ open class BrickCollectionView: UICollectionView {
     private func isBrickRegistered(brick: Brick) -> Bool {
         return identifier(for: brick) != nil
     }
+
+    open override func setContentOffset(_ contentOffset: CGPoint, animated: Bool) {
+        super.setContentOffset(contentOffset, animated: animated)
+        self.layoutIfNeeded()
+    }
 }
 
 extension BrickCollectionView: AsynchronousResizableDelegate {
 
-    public func performResize(cell: BrickCell, completion: ((Bool) -> Void)? = nil) {
+    public func performResize(cell: BrickCell, completion: @escaping (() -> Void)) {
+        assert(Thread.isMainThread)
         guard let indexPath = self.indexPath(for: cell) else {
-            completion?(true)
+            // If the cell is offscreen, we can force the height by setting the frame
+            let height = cell.heightForBrickView(withWidth: cell.frame.width)
+            if height != cell.frame.height {
+                cell.frame = CGRect(origin: cell.frame.origin, size: CGSize(width: cell.frame.size.width, height: height))
+                self.collectionViewLayout.invalidateLayout()
+            }
+            completion()
             return
         }
         let newHeight = cell.heightForBrickView(withWidth: cell.frame.width)
-        weak var weakLayout = self.layout
         self.performBatchUpdates({
-            weakLayout?.updateHeight(indexPath, newHeight: newHeight)
+            self.layout.updateHeight(indexPath, newHeight: newHeight)
```

Comment:
This will cause a memory leak

Reply:
I'm not sure if it will. Doesn't the collectionView (self) always have a strong reference to to the layout anyways?  I can change this back, doesn't hurt to leave it as is.

Reply:
updated